_source/_posts/2022-09-26-micro-frontends-java-microservices.adoc

```diff
@@ -0,0 +1,625 @@
+
+image::{% asset_path 'blog/micro-frontends-java/jhipster-microfrontends.png' %}[alt=JHipster microfrontends architecture,width=800,align=center]
+
+You might notice I spelled micro frontends three different ways in the previous paragraph. The current literature is all over the place on this one!
+
+// https://twitter.com/mraible/status/1499098253920460802
+++++
+{% twitter 1499098253920460802 %}
+++++
+
+I'm going to use "micro frontends" for the remainder of this post since that's what https://twitter.com/thecamjackson[Cam Jackson] used in his https://martinfowler.com/articles/micro-frontends.html[Micro Frontends article] on Martin Fowler's blog.
+
+== A quick introduction to Module Federation
+
+Webpack's https://webpack.js.org/concepts/module-federation/[Module Federation] is one of the best known implementations for micro frontends. Its https://webpack.js.org/plugins/module-federation-plugin[`ModuleFederationPlugin`] allows a build to provide or consume modules with other independent builds at runtime. It even allows you to share libraries between frontends to reduce the size of remote bundles.
+
+https://twitter.com/ScriptedAlchemy[Zack Jackson] is the creator of Module Federation and recently https://twitter.com/ScriptedAlchemy/status/1564411584851505153[collaborated] with https://twitter.com/ManfredSteyer[Manfred Steyer] to create Native Federation. This means you can use micro frontend concepts with any build tool, not just webpack.
+
+// https://twitter.com/ManfredSteyer/status/1564312149580582912
+++++
+{% twitter 1564312149580582912 %}
+++++
+
+== Why should Java developers care?
+
+I think micro frontends an excellent architecture concept. With microservices, you weave everything together on the backend with protocols like HTTP, gRPC, and GraphQL. With micro frontends, it's all HTTP. You can literally see your app get stitched together by watching your browser's network console and seeing remote modules load.
+
+I've encountered quite a few monolith UIs in my time as a consultant. The backend was a beautiful microservice architecture, but it was all coupled together on the frontend. There's a good chance a lot of Java developers don't care about the UI because they just work on the beautiful backends. However, if you consider yourself a Java _web_ developer, micro frontends are as revolutionary as HTML5!
+
+And that's the beauty of this tutorial, you don't have to write any micro frontends. JHipster can create them for you!
+
+== Micro frontends in action with JHipster
+
+I used a JDL (JHipster Domain Language) file named `reactive-ms.jdl` to create the reactive Java microservices tutorial I mentioned earlier. You can see this file online in the https://github.com/jhipster/jdl-samples/blob/main/reactive-ms.jdl[JDL samples repository].
+
+I copied this file to `reactive-mf.jdl` and changed a few things for this tutorial:
```

Comment:
may be worth mentioning the `jhipster download reactive-ms.jdl` command here as well?

Reply:
I mention this below after they've created the `micro-frontends-jhipster` directory.